File: internal/model/game.go
package model

type Game struct {
	score gameScore
}

type gameScore struct {
	BlueScore  int `json:"blueScore"`
	WhiteScore int `json:"whiteScore"`
}

func (gS *gameScore) ResetScore() {
	gS.BlueScore = 0
	gS.WhiteScore = 0
}

func (g *Game) AddGoal(teamID int) {
	switch teamID {
	case 1:
		g.score.WhiteScore++
	case 2:
		g.score.BlueScore++
	}
}

File: frontend/smart-kickers-game/models/game.js
class Game {
  constructor(blueScore, whiteScore, blueAdded, whiteAdded) {
    this.blueScore = blueScore;
    this.whiteScore = whiteScore;
    this.whiteAdded = whiteAdded;
    this.blueAdded = blueAdded;
  }
  get blueScore() {
    return this.blueScore;
  }
  set blueScore(blueScore) {
    this.blueScore = blueScore;
  }
  get whiteScore() {
    return this.whiteScore;
  }
  set whiteScore(whiteScore) {
    this.whiteScore = whiteScore;
  }
  get blueAdded() {
    return this.blueAdded;
  }
  set blueAdded(blueAdded) {
    this.blueAdded = blueAdded;
  }
  get whiteAdded() {
    return this.whiteAdded;
  }
  set whiteAdded(whiteAdded) {
    this.whiteAdded = whiteAdded;
  }
}
